thetis/solver2d.py

```diff
@@ -574,25 +608,24 @@ def create_timestepper(self):
             assert self.options.timestepper_type in steppers
         except AssertionError:
             raise Exception('Unknown time integrator type: {:s}'.format(self.options.timestepper_type))
-        if self.options.solve_tracer:
-            try:
-                assert self.options.timestepper_type not in ('PressureProjectionPicard', 'SSPIMEX', 'SteadyState')
-            except AssertionError:
-                raise NotImplementedError("2D tracer model currently only supports SSPRK33, ForwardEuler, BackwardEuler, DIRK22, DIRK33 and CrankNicolson time integrators.")
-            self.timestepper = coupled_timeintegrator_2d.CoupledMatchingTimeIntegrator2D(
-                weakref.proxy(self), steppers[self.options.timestepper_type],
-            )
-        elif self.options.sediment_model_options.solve_suspended_sediment or self.options.sediment_model_options.solve_exner:
-            try:
-                assert self.options.timestepper_type not in ('PressureProjectionPicard', 'SSPIMEX', 'SteadyState')
-            except AssertionError:
-                raise NotImplementedError("2D sediment model currently only supports SSPRK33, ForwardEuler, BackwardEuler, DIRK22, DIRK33 and CrankNicolson time integrators.")
+        use_sediment_model = self.options.sediment_model_options.solve_suspended_sediment or self.options.sediment_model_options.solve_exner
+        solve_nh_pressure = self.options.nh_model_options.solve_nonhydrostatic_pressure
+        if self.options.solve_tracer or use_sediment_model or solve_nh_pressure:
+            assert self.options.timestepper_type not in ('PressureProjectionPicard', 'SSPIMEX', 'SteadyState'), \
+                "2D model with tracer, nh pressure or sediments currently only supports SSPRK33, ForwardEuler, BackwardEuler"
```

Comment:
I have corrected these. Actually NH is fully related to the original SWE system. All timeintegrators should be compatible with NH, since the hydrostatic solution is obtained first and then corrected by NH pressure (Poisson eq) and final free surface correction.